File: Love-Babbar-450-In-CSharp/04_searching_and_sorting/05_max_min_with_min_comparison.cs
using System;
using System.Collections.Generic;
using System.Text;
using Xunit;

namespace _04_searching_and_sorting
{
    public class _05_max_min_with_min_comparison
    {
        [Fact]
        public void reverse_arrayTest() { }


        /*
            link: https://practice.geeksforgeeks.org/problems/middle-of-three2926/1
        */


        // ----------------------------------------------------------------------------------------------------------------------- //
        /*
            TC: O(1)
        */
        private int middle1(int A, int B, int C)
        {
            //code here//Position this line where user code will be pasted.
            int sum = A + B + C;
            //sum -= min({ A, B, C}) +max({ A, B, C});
            return sum;
        }


        // ----------------------------------------------------------------------------------------------------------------------- //
        /*
            TC: O(1)
        */
        private int middle2(int A, int B, int C)
        {
            if (A < B)
            {
                return (A > C) ? A : Math.Min(C, B);
            }
            return (A < C) ? A : Math.Max(B, C);
        }


        // ----------------------------------------------------------------------------------------------------------------------- //
        /*
            similar solution as above.
            TC: O(1)
        */
        private int middle3(int A, int B, int C)
        {
            if (A < B)
            {
                return (B < C) ? B : Math.Max(A, C);
            }
            return (A < C) ? A : Math.Max(B, C);
        }

    }
}

File: Love-Babbar-450-In-CPPTest/04_searching_and_sorting/05_max_min_with_min_comparison.cpp
/*
    link: https://practice.geeksforgeeks.org/problems/middle-of-three2926/1
*/
#include "CppUnitTest.h"
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;
using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace LoveBabbar450InCPPTest
{
    TEST_CLASS(SearchSort)
    {
    public:

        TEST_METHOD(find1Test)
        {
            std::string charM = "harhs";
            int age = 14;
            age = 55;
            std::string lastName = "<<charM <<singh";

        }
    };
}

// ----------------------------------------------------------------------------------------------------------------------- //
/*
    TC: O(1)
*/
int middle(int A, int B, int C) {
        //code here//Position this line where user code will be pasted.
    int sum = A + B + C;
    sum -= min({ A, B, C }) + max({ A, B, C });
    return sum;
}


// ----------------------------------------------------------------------------------------------------------------------- //
/*
    TC: O(1)
*/
int middle2(int A, int B, int C) {
    if (A < B) {
        return (A > C) ? A : min(C, B);
    }
    return (A < C) ? A : max(B, C);
}


// ----------------------------------------------------------------------------------------------------------------------- //
/*
    similar solution as above.
    TC: O(1)
*/
int middle3(int A, int B, int C) {
    if (A < B)
        return (B < C) ? B : max(A, C);
    return (A < C) ? A : max(B, C);
}